subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/UploadToQapps.kt

```diff
@@ -19,13 +22,19 @@ class UploadToQapps(context: String, artifactsConfiguration: ArtifactsConfigurat
             "${this.javaClass.simpleName} check requires $plugin being applied to project"
         }
 
+        val uploadCondition = UploadToQappsCondition(project.cdBuildConfig.orNull)
+
         return artifactsMap.map { (_, output) ->
             when (output) {
                 is Output.ApkOutput ->
                     project.tasks.qappsTaskProvider(output.variantName)
                         .apply {
                             configure {
                                 it.getApk().set(File(output.path))
+                                it.releaseChain.set(uploadCondition.isReleaseChain())
+                                it.onlyIf {
```

Comment:
We can simply don't create a task and remove `onlyIf clause`

Reply:
I prefer `onlyIf` in similar cases due to easier diagnostics in case of problems.
Let's imagine a typical bug: I don't see artifacts in Artifactory. The first thing to check is the task. I'll see that this task is skipped due to `onlyIf` clause. I know the reason and there to find it.
When I don't see the task, I have to check more code that creates it.

Reply:
Interesting point, I agree. Let write tests on that scenario

Reply:
Added `UploadToQappsTest`